smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaThrottledLatestProcessedCommit.java

```diff
@@ -38,7 +38,7 @@
 
     private static final Map<String, Map<Integer, TopicPartition>> TOPIC_PARTITIONS_CACHE = new ConcurrentHashMap<>();
 
-    private final Map<TopicPartition, OffsetStore> offsetStores = new HashMap<>();
+    private final Map<TopicPartition, OffsetStore> offsetStores = new ConcurrentHashMap<>();
```

Comment:
This could be a problem as OffsetStore is not thread safe and was designed to be called within the context thread.

In `partitionsRevoked` we call `clearLesserSequentiallyProcessedOffsetsAndReturnLargestOffset` which mutates the OffsetStore. Is it not possible to make a blocking call in the context thread from the poll thread?

Reply:
I ran the tests locally and they passed. What I did was to revert the `offsetStores` map back to a HashMap and ran the relevant parts of `partitionsRevoked` within the context. I'll post my changes later.

Reply:
no partitionRevoked must run completely from the polling thread. It cannot run from the event loop. When the method returns, everything must have been completed or failed. 

One thing I was thinking is to block the polling thread and use a countdown latch decreased from the event loop. However, it can introduce deadlocks.

Reply:
I just used a FutureTask to manipulate the OffsetStore and fed it into the context and waited for a response on the poll thread. Seemed to work. Going to do some more tests.

Reply:
Here what I tested
https://github.com/pcasaes/smallrye-reactive-messaging/commit/3db01661479eb0f3404adb921a5cb8d7c8c66488

Notice there are a couple of FIXMEs.

I did the run the tests as well as manual testing. Worked well.

I did pickup a small bug which I'll open a separate PR for. During revoke we should be committing the offsets of those revoked partitions. There's a `not contains` which should `contains`.

Reply:
I've rebased the branch and updated to fit the synchronization protocol.
The FutureTask is a great idea!